web/crux/src/services/kratos.service.ts

```diff
@@ -120,25 +120,48 @@ export default class KratosService {
   }
 
   async enableOnboarding(userId: string): Promise<void> {
+    const identity = (
+      await this.identity.getIdentity({
+        id: userId,
+      })
+    ).data
+
+    // Due to a bug in kratos (https://github.com/ory/kratos/pull/3304)
+    // the whole metadata needs to be updated
     await this.identity.patchIdentity({
       id: userId,
       jsonPatch: [
         {
-          op: 'remove',
-          path: '/metadata_public/disableOnboarding',
+          op: 'add',
+          path: '/metadata_public',
+          value: {
+            ...identity.metadata_public,
+            disableOnboarding: undefined,
+          },
         },
       ],
     })
   }
 
   async disableOnboarding(userId: string): Promise<void> {
+    const identity = (
+      await this.identity.getIdentity({
+        id: userId,
+      })
+    ).data
+
+    // Due to a bug in kratos (https://github.com/ory/kratos/pull/3304)
```

Comment:
This is fixed in here: https://github.com/ory/kratos/issues/3244 - so is this still relevant?

Reply:
Talked in person.